backend/app/src/main/java/com/moh/phlat/backend/controller/ProcessDataController.java

```diff
@@ -392,13 +448,215 @@ public ResponseEntity<ResponseMessage> plrLoad(@PathVariable Long controlTableId
 		return ResponseEntity.status(HttpStatus.OK).body(new ResponseMessage("success", 200, "PLR load process started!",_control));
 	}
 
-@PreAuthorize("hasAnyRole(@roleService.getAllRoles())")
-@GetMapping("/reportsummary/{controlTableId}")
-public @ResponseBody ResponseEntity<ResponseMessage> getReportSummaryByControlTableId(
-		@PathVariable Long controlTableId, @RequestParam(required = false) String rowStatus) {
-	
-	List<ReportSummary> _list = dbUtilityService.getReportSummary(controlTableId);
+	@PreAuthorize("hasAnyRole(@roleService.getAllRoles())")
+	@GetMapping("/reportsummary/{controlTableId}")
+	public @ResponseBody ResponseEntity<ResponseMessage> getReportSummaryByControlTableId(
+			@PathVariable Long controlTableId, @RequestParam(required = false) String rowStatus) {
+		
+		List<ReportSummary> _list = dbUtilityService.getReportSummary(controlTableId);
+		
+		return ResponseEntity.status(HttpStatus.OK).body(new ResponseMessage("success", 200, "", _list));
+	}
 	
-	return ResponseEntity.status(HttpStatus.OK).body(new ResponseMessage("success", 200, "", _list));
+	@PreAuthorize("hasAnyRole(@roleService.getAllRoles())")
+	@GetMapping("/getColumnList/{controlTableId}")
+	public ResponseEntity<ResponseMessage> getColumnLists(@PathVariable Long controlTableId, @RequestParam(required = false) String columnKey) {
```

Comment:
any reason why columnKey is optional?

Reply:
Will change.  Don't remember what my thought process was, may have been a copy past forget modify issue.